src/main.ts

```diff
@@ -31,26 +42,36 @@ export async function run(): Promise<void> {
       }
     }
 
-    // Fork the specified repo into user namespace, unless an organization is specified
-    core.info(`⑂ Creating fork of repository ${repo}...`)
-    await forkRepo(owner, repo, org)
-
-    // Optionally check org membership status for a specified user, and invite if missing
-    if (addUser && org && typeof user !== 'undefined') {
+    // Optionally enforce organization membership status for a specified user
+    // Exit early if the user is determined not to be an organization member
+    if (checkUser && org && typeof user !== 'undefined') {
       core.info(
         `🔍 Checking membership status of user ${user} in ${org} organization...`
       )
       if (await isOrgMember(org, user)) {
         core.info(
-          `✅ User ${user} already a member of ${org}, no action needed`
+          `✅ User ${user} is a member of ${org}, proceeding with fork request...`
         )
       } else {
-        core.info(
-          `📥 Inviting user ${user} to ${org} org, make sure they check their inbox!`
+        core.setFailed(
+          `🚨 User ${user} not a member of ${org}, please join the organization before trying again!`
         )
-        inviteMember(org, user)
+        // Do not proceed with fork creation if org membership check fails
+        return
       }
     }
+
+    // Fork the specified repo into user namespace, unless an organization is specified
+    core.info(`⑂ Creating fork of repository ${repo}...`)
+    await forkRepo(owner, repo, org)
+
+    // Optionally promote the requesting user's permissions to admin for the forked repository
+    if (promoteUser && org && typeof user !== 'undefined') {
+      core.info(
+        `⏫ Promoting user permissions for ${user} to ${PERMISSIONS.ADMIN}`
```

Comment:
Non-blocker:  a "nice to have" for a future release might be to allow to set the permission from the list of available permissions vs. just ADMIN.

Reply:
Definitely agree! Part of my thinking in creating a `PERMISSIONS` const now was to enable our ability to do this in the future, if desired. For the scope of this refactor, though, I'd prefer to keep `admin` hardcoded initially, since support for user permissions was already a nice-to-have :)